### PR TITLE
fix(avoidance): the module ignored merging objects unexpectedly

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Related PR: https://github.com/autowarefoundation/autoware_launch/pull/923
Related ticket: https://tier4.atlassian.net/browse/RT1-202

Fix bug. The module ignored merging vehicle unexpectedly. Originally, I designed the module generates avoidance path for every objects that are NOT on ego lane. (Only for objects in intersection, the module sometimes ignores them.)

https://github.com/autowarefoundation/autoware.universe/assets/44889564/a436e96e-7760-4668-b32c-3c60a05dbd91

And, this PR includes small refactoring as well.

## Tests performed

```
webauto ci scenario run --project-id prd_jt --scenario-id f9e88ceb-0e06-48d5-9bad-be5ed4a244a0 --scenario-parameters '__tier4_modifier_npc_position_t=3'
```

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/9f0e732d-ceea-531e-9496-22d173dcca85?project_id=prd_jt)

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by shmpwk** [comment]
@Owen-Liuyuxuan Could you review this code? Ota-san will test scenario so please just check the code.

**Comment by Owen-Liuyuxuan** [comment]
Thanks for the great response and contribution. The refactoring improves the readability of the codes; the codes update the target object filtering logic which should correctly implement the logic we hope to have.

I notice that `overhang_lanelet` array is added to the debug marker arrays, could there be additional materials/documentation on what we can see with the new debug markers?

File: planning/behavior_path_avoidance_module/src/debug.cpp [hunk 1]
```diff
@@ -188,10 +188,25 @@ MarkerArray createObjectInfoMarkerArray(const ObjectDataArray & objects, std::st
   return msg;
 }
 
+MarkerArray createOverhangLaneletMarkerArray(const ObjectDataArray & objects, std::string && ns)
```
Comment (Owen-Liuyuxuan) [inline planning/behavior_path_avoidance_module/src/debug.cpp:191]: Will there be additional documentation or image showing the usage of this marker?

Comment (satoshi-ota) [reply planning/behavior_path_avoidance_module/src/debug.cpp:191]: @Owen-Liuyuxuan Thank you for your comment.

Unfortunatelly, there is no document about overhang lanelet. It is one of the ego driving lanelets (closest lanelet to the object) and used in the logic to check whether the object is on the ego lane.

Actually, it is difficult to understand without any comment. So, I'll add above brief explanation on code.

**Comment by satoshi-ota** [comment]
Please check https://github.com/autowarefoundation/autoware_launch/pull/923 as well.

File: planning/behavior_path_avoidance_module/src/debug.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_avoidance_module/src/debug.cpp:191]: @Owen-Liuyuxuan I updated comment in code in https://github.com/autowarefoundation/autoware.universe/pull/6601/commits/1344e6c9295eaf6feffe0dc6b83659d5435a2705.

**Review by shmpwk — APPROVED** [review]
LGTM

**Pull request merged**